radicale/storage.py

```diff
@@ -621,43 +689,150 @@ def move(cls, item, to_collection, to_href):
         if item.collection._filesystem_path != to_collection._filesystem_path:
             cls._sync_directory(item.collection._filesystem_path)
 
+    @classmethod
+    def _clean_cache(cls, folder, names, max_age=None):
+        # Delete all ``names`` in ``folder`` that are older than ``max_age``.
+        age_limit = time.time() - max_age if max_age is not None else None
+        modified = False
+        for name in names:
+            if not is_safe_filesystem_path_component(name):
+                continue
+            if age_limit is not None:
+                try:
+                    # Race: Another process might have deleted the file.
+                    mtime = os.path.getmtime(os.path.join(folder, name))
+                except FileNotFoundError:
+                    continue
+                if mtime > age_limit:
+                    continue
+            cls.logger.debug("Found expired item in cache: %s", name)
+            # Race: Another process might have deleted or locked the
+            # file.
+            try:
+                os.remove(os.path.join(folder, name))
+            except (FileNotFoundError, PermissionError):
+                continue
+            modified = True
```

Comment:
> Shorter and more focused on on what should be done here

It's not equivalent. The file is not removed if ``age_limit`` is ``None``.

> compute path only once for speed

Maybe, but it shouldn't make a measurable difference.